shai/src/main/scala/at/forsyte/apalache/shai/RpcServer.scala

```diff
@@ -15,7 +15,10 @@ object RpcServer extends ServerMain {
     // A thread safe mutable reference to the active connections
     // See https://zio.dev/version-1.x/datatypes/concurrency/ref/
     connections <- Ref.make(Map[UUID, Conn]())
-  } yield new TransExplorerService(connections)
+    // Ensure atomic access to the parser
+    // See https://github.com/informalsystems/apalache/issues/1114#issuecomment-1180534894
+    parserSemaphore <- Semaphore.make(permits = 1)
```

Comment:
Somewhat orthogonal, but I'm wondering if we shouldn't sequentialize parsing once and for all in `SanyImporter`?